File: Midterm_example/src/main/java/extractInterface/After.java
package extractInterface;

/**
 * - Call new interface Billable
 * - Declare getRate and hasSpecialSkill in interface
 * - Declare Employee class as implement interface Billable
 */
interface Billable	{
    interface int getRate();
    interface boolean hasSpecialSkill();
}
class Employee implements Billable	{
    String getName();
    String getDepartment();
    int getRate();
    boolean hasSpecialSkill();
}

class Client{
    /**
     * Change paramenter from Employee to Billable
     */
    double charge(Billable bill, int days){
        int base = bill.getRate() * days;

        if(bill.hasSpecialSkill())
            return base * 1.08;
        else return base;
    }
}

File: CSharpGenelization 3/CSharpGenelization/ExtractInterface/After.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpGenelization.ExtractInterface
{
    internal class After
    {
        
         //- Call new interface Billable
         //- Declare getRate and hasSpecialSkill in interface
         //- Declare Employee class as implement interface Billable
         
        interface Billable
        {
            int getRate();
            bool hasSpecialSkill();
        }
        class Employee : Billable
        {
            public int getRate() { }
            public bool hasSpecialSkill() { }
            public String getName() { }
            public String getDepartment() { }
        }

    class Client
    {

        //Change paramenter from Employee to Billable
        double charge(Billable bill, int days)
        {
            int basic = bill.getRate() * days;

            if (bill.hasSpecialSkill())
                return basic * 1.08;
            else return basic;
        }
    }
}
}
